lib/Basics/SupervisedBuffer.h

```diff
@@ -12,12 +35,31 @@ class SupervisedBuffer : public Buffer<uint8_t> {
 
   explicit SupervisedBuffer(arangodb::ResourceMonitor& monitor)
       : _usageScope{monitor, 0} {
+    // account the initial inline capacity one time (192 bytes)
     _usageScope.increase(this->capacity());
   }
 
-  uint8_t* stealWithMemoryAccounting(ResourceUsageScope& owningScope) noexcept {
-    owningScope.increase(_usageScope.trackedAndSteal());
+  uint8_t* stealWithMemoryAccounting(ResourceUsageScope& owningScope) {
+    std::size_t tracked = 0;
+    // the buffer's usage scope is gonna account for 0 bytes now and give the
+    // value it accounted for to the owning scope, so the amount of memory that
+    // the buffer allocated which was accounted by its scope is now gonna be
+    // accounted by the owning scope. If it throws when we increase the owning
+    // scope, we don't need to increase _usageScope again, because decrease
+    // didn't happen
```

Comment:
```suggestion
    // accounted by the owning scope.
    // We do not have an atomic operation here, so we will first decrease on current scope
    // and hand it over to the new scope. We did not do it the other way round, as this would double
    // the reported memory for a short time, which could cause an perfectly fine operation to OOM.
    // The memory is never used twice.
```